### PR TITLE
docs: add details on how to use custom_build with buildah

### PR DESCRIPTION
**Pull request by nicks**
Hello @jazzdan, @maiamcc,

Please review the following commits I made in branch nicks/custombuild:

6e2fd2d04666987d7c4b7f09d7b896934c197ba0 (2019-11-26 15:00:40 -0500)
docs: add details on how to use custom_build with buildah

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -514,6 +514,7 @@ def custom_build(ref: str, command: str, deps: List[str], tag: str = "", disable
     tag: Some tools can't change the image tag at runtime. They need a pre-specified tag. Tilt will set ``$EXPECTED_REF = image_name:tag``,
        then re-tag it with its own tag before pushing to your cluster. See `the bazel guide <integrating_bazel_with_tilt.html>`_ for an example.
     disable_push: whether Tilt should push the image in to the registry that the Kubernetes cluster has access to. Set this to true if your command handles pushing as well.
+    skips_local_docker: Whether your builder write the image to your local Docker image store. Set this to true if you're using a cloud-based builder or independent image builder like ``buildah``.
```
Comment (maiamcc) [inline api/api.py]: "whether your builder writes" or "builder will write".

File: docs/custom_build.md [hunk 2]
```diff
@@ -20,9 +34,56 @@ custom_build(
 )
 ```
 
-One piece of this needs more explanation: `$EXPECTED_REF`.
+Tilt will run this command to build the image, verify that the image is in Docker,
```
Comment (maiamcc) [inline docs/custom_build.md]: > verify that the image is in Docker
imo a little clearly to say something like: "verify that the expected image exists in your local Docker image registry, and (if necessary) push the image..."

(Idk if "local Docker image registry" is the right verbiage but you get the idea)

File: docs/custom_build.md [hunk 3]
```diff
@@ -20,9 +34,56 @@ custom_build(
...
+then push the image to the appropriate image registry.
+
+You can also use this to use `docker` flags that the `docker_build()` function doesn't support.
```
Comment (maiamcc) [inline docs/custom_build.md]: "...also use this pattern..." OR "...also invoke `docker build` via `custom_build` to use flags that..."

File: docs/custom_build.md [hunk 4]
```diff
@@ -20,9 +34,56 @@ custom_build(
...
+
+### Buildah (or any Image Builder Independent of Docker)
```
Comment (maiamcc) [inline docs/custom_build.md]: maybe don't title case this so it's clear what the proper nouns are? i don't feel strongly tho 🤷‍♀

File: docs/custom_build.md [hunk 5]
```diff
@@ -20,9 +34,56 @@ custom_build(
...
+
+- They often require privileged access. You may need to run tilt with `sudo` or
+  inside an appropriate sandbox
+
+- If you're using Tilt to push to an insecure registry, you will need to
+  configure your builder for that registry. For example, to use `buildah` with
+  Microk8s, you need to add `localhost:32000` to
+  [registries.insecure](https://github.com/containers/buildah/blob/master/install.md#registriesconf)
+
+### Bazel (or any Image Builder that Interoperates with Docker)
```
Comment (maiamcc) [inline docs/custom_build.md]: see above about title casing

File: docs/custom_build.md [hunk 6]
```diff
@@ -20,9 +34,56 @@ custom_build(
...
+  'frontend',
+  'buildah bud -t $EXPECTED_REF frontend && buildah push $EXPECTED_REF $EXPECTED_REF',
+  ['./frontend'],
+  skips_local_docker=True,
+)
+```
+
+The `skips_local_docker` parameter tells Tilt that we don't expect the image to
+ever show up in the local Docker image store, so it won't be able to verify the
+image locally.
```
Comment (maiamcc) [inline docs/custom_build.md]: This maybe reads a little better:
> The `skips_local_docker` parameter indicates that we don't expect the image to ever show up in the local Docker image store, so Tilt shouldn't try to verify the image locally.

File: docs/custom_build.md [hunk 7]
```diff
@@ -20,9 +34,56 @@ custom_build(
...
+
+There are a couple of caveats you should be aware of with `buildah` (and similar builders):
+
+- They often require privileged access. You may need to run tilt with `sudo` or
```
Comment (maiamcc) [inline docs/custom_build.md]:
```suggestion
- They often require privileged access. You may need to run Tilt with `sudo` or
```

File: docs/custom_build.md (hunk 5 above)
Comment (maiamcc) [reply docs/custom_build.md]: but also, this seems like a simpler case than `buildah` et al, I might put this section above "image builders independent of docker" so the doc progresses from most dockery --> least.

File: api/api.py (hunk 1 above)
Comment (nicks) [reply api/api.py]: done

File: docs/custom_build.md (hunk 2 above)
Comment (nicks) [reply docs/custom_build.md]: done

File: docs/custom_build.md (hunk 3 above)
Comment (nicks) [reply docs/custom_build.md]: done

File: docs/custom_build.md (hunk 4 above)
Comment (nicks) [reply docs/custom_build.md]: done

File: docs/custom_build.md (hunk 5 above)
Comment (nicks) [reply docs/custom_build.md]: done

File: docs/custom_build.md (hunk 6 above)
Comment (nicks) [reply docs/custom_build.md]: done

File: docs/custom_build.md (hunk 7 above)
Comment (nicks) [reply docs/custom_build.md]: done

**Comment by nicks** [comment]
good comments, thanks!

**Pull request merged**